<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>Sharing Annotated IFCB Images</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Sharing Annotated IFCB Images</h1>



<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>Annotated images can be shared as zipped <code>.png</code> packages
through various data repositories (e.g., Kraft et al., 2022; Torstensson
et al., 2024), enabling others to train or enhance their image
classifiers. This vignette provides a step-by-step guide to extracting
and preparing such images for publication using the <code>iRfcb</code>
package.</p>
<p>The workflow assumes that Regions of Interest (ROIs) have been
annotated using the MATLAB code from the <a href="https://github.com/hsosik/ifcb-analysis">ifcb-analysis</a>
repository (Sosik and Olson, 2007). However, the methods presented can
be adapted to process images generated by other software platforms. The
archive can be shared through various sources, such as <a href="https://figshare.com/">Figshare</a>, <a href="https://zenodo.org/">Zenodo</a>, <a href="https://b2share.eudat.eu/">EUDAT</a>. Links to some repositories
from Northern Europe are gathered at the <a href="https://nordicmicroalgae.org/annotated-images/">Nordic
Microalgae</a> webpage. Images may also be shared through EcoTaxa, which
is demonstrated in the <a href="../articles/ecotaxa-tutorial.html">Prepare IFCB Images for
EcoTaxa</a> tutorial.</p>
<p>Additionally, this vignette shows how users of the <a href="https://github.com/hsosik/ifcb-analysis">ifcb-analysis</a> package
can share and merge multiple datasets of manually annotated images,
enabling MATLAB users to incorporate external datasets into their random
forest algorithms.</p>
</div>
<div id="getting-started" class="section level2">
<h2>Getting Started</h2>
<div id="installation" class="section level3">
<h3>Installation</h3>
<p>You can install the package from GitHub using the
<code>remotes</code> package:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="co"># install.packages(&quot;remotes&quot;)</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a>remotes<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;EuropeanIFCBGroup/iRfcb&quot;</span>)</span></code></pre></div>
<p>Load the <code>iRfcb</code> library:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">library</span>(iRfcb)</span></code></pre></div>
</div>
<div id="download-sample-data" class="section level3">
<h3>Download Sample Data</h3>
<p>To get started, download sample data from the <a href="https://doi.org/10.17044/scilifelab.25883455.v3">SMHI IFCB
Plankton Image Reference Library</a> (Torstensson et al. 2024) with the
following function:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="co"># Define data directory</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>data_dir <span class="ot">&lt;-</span> <span class="st">&quot;data&quot;</span></span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="co"># Download and extract test data in the data folder</span></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a><span class="fu">ifcb_download_test_data</span>(<span class="at">dest_dir =</span> data_dir,</span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>                        <span class="at">max_retries =</span> <span class="dv">10</span>,</span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a>                        <span class="at">sleep_time =</span> <span class="dv">30</span>,</span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a>                        <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</div>
</div>
<div id="extract-annotated-images" class="section level2">
<h2>Extract Annotated Images</h2>
<p>Extract annotated ROIs as <code>.png</code> images in subfolders for
each class, skipping the <strong>unclassified</strong> (class id 1)
category:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="co"># Extract .png images</span></span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="fu">ifcb_extract_annotated_images</span>(<span class="at">manual_folder =</span> <span class="st">&quot;data/manual&quot;</span>,</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a>                              <span class="at">class2use_file =</span> <span class="st">&quot;data/config/class2use.mat&quot;</span>,</span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a>                              <span class="at">roi_folders =</span> <span class="st">&quot;data/data&quot;</span>,</span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a>                              <span class="at">out_folder =</span> <span class="st">&quot;data/extracted_images&quot;</span>,</span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a>                              <span class="at">skip_class =</span> <span class="dv">1</span>, <span class="co"># or &quot;unclassified&quot;</span></span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a>                              <span class="at">verbose =</span> <span class="cn">FALSE</span>) <span class="co"># Do not print messages</span></span></code></pre></div>
</div>
<div id="package-png-directory" class="section level2">
<h2>Package PNG Directory</h2>
<p>Prepare the PNG directory for publication as a zip-archive, similar
to the files in the <a href="https://doi.org/10.17044/scilifelab.25883455">SMHI IFCB Plankton
Image Reference Library</a> (Torstensson et al. 2024). This function
reads, updates, and incorporates a <strong>README</strong> file into the
zip archive. A template <strong>README</strong> file is included with
the <code>iRfcb</code> package.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="co"># Create zip-archive</span></span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="fu">ifcb_zip_pngs</span>(<span class="at">png_folder =</span> <span class="st">&quot;data/extracted_images&quot;</span>,</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a>              <span class="at">zip_filename =</span> <span class="st">&quot;data/zip/ifcb_annotated_images_corrected.zip&quot;</span>,</span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a>              <span class="at">readme_file =</span> <span class="fu">system.file</span>(<span class="st">&quot;exdata/README-template.md&quot;</span>, </span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a>                                        <span class="at">package =</span> <span class="st">&quot;iRfcb&quot;</span>), <span class="co"># Template icluded in `iRfcb`</span></span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a>              <span class="at">email_address =</span> <span class="st">&quot;tutorial@test.com&quot;</span>,</span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a>              <span class="at">version =</span> <span class="st">&quot;1.1&quot;</span>,</span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a>              <span class="at">print_progress =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>## Creating README file...
## Creating MANIFEST.txt...
## Creating zip archive...
## Zip archive created successfully: /private/var/folders/7t/9x250txj7hz4v60rzqtpn9jm0000gn/T/Rtmp4sQWoM/Rbuildf6a31bdf4361/iRfcb/vignettes/data/zip/ifcb_annotated_images_corrected.zip</code></pre>
</div>
<div id="package-matlab-directory" class="section level2">
<h2>Package MATLAB Directory</h2>
<p>Prepare the MATLAB directory for publication as a zip-archive,
similar to the files in the <a href="https://doi.org/10.17044/scilifelab.25883455">SMHI IFCB Plankton
Image Reference Library</a>:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="co"># Create zip-archive</span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a><span class="fu">ifcb_zip_matlab</span>(<span class="at">manual_folder =</span> <span class="st">&quot;data/manual&quot;</span>,</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a>                <span class="at">features_folder =</span> <span class="st">&quot;data/features&quot;</span>,</span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a>                <span class="at">class2use_file =</span> <span class="st">&quot;data/config/class2use.mat&quot;</span>,</span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a>                <span class="at">zip_filename =</span> <span class="st">&quot;data/zip/ifcb_matlab_files_corrected.zip&quot;</span>,</span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a>                <span class="at">data_folder =</span> <span class="st">&quot;data/data&quot;</span>,</span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a>                <span class="at">readme_file =</span> <span class="fu">system.file</span>(<span class="st">&quot;exdata/README-template.md&quot;</span>, </span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a>                                          <span class="at">package =</span> <span class="st">&quot;iRfcb&quot;</span>), <span class="co"># Template icluded in `iRfcb`</span></span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a>                <span class="at">matlab_readme_file =</span> <span class="fu">system.file</span>(<span class="st">&quot;exdata/MATLAB-template.md&quot;</span>, </span>
<span id="cb7-10"><a href="#cb7-10" tabindex="-1"></a>                                                 <span class="at">package =</span> <span class="st">&quot;iRfcb&quot;</span>), <span class="co"># Template icluded in `iRfcb`</span></span>
<span id="cb7-11"><a href="#cb7-11" tabindex="-1"></a>                <span class="at">email_address =</span> <span class="st">&quot;tutorial@test.com&quot;</span>,</span>
<span id="cb7-12"><a href="#cb7-12" tabindex="-1"></a>                <span class="at">version =</span> <span class="st">&quot;1.1&quot;</span>,</span>
<span id="cb7-13"><a href="#cb7-13" tabindex="-1"></a>                <span class="at">print_progress =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>## Listing all files...
## Copying manual files...
## Copying feature files...
## Copying data files...
## Copying class2use file...
## Creating README file...
## Creating MANIFEST.txt...
## Creating zip archive...
## Zip archive created successfully: /private/var/folders/7t/9x250txj7hz4v60rzqtpn9jm0000gn/T/Rtmp4sQWoM/Rbuildf6a31bdf4361/iRfcb/vignettes/data/zip/ifcb_matlab_files_corrected.zip</code></pre>
</div>
<div id="create-manifest.txt" class="section level2">
<h2>Create MANIFEST.txt</h2>
<p>Create a manifest file for the zip-archive (required for some data
repositories):</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="co"># Create MANIFEST.txt of the zip folder content</span></span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a><span class="fu">ifcb_create_manifest</span>(<span class="st">&quot;data/zip&quot;</span>)</span></code></pre></div>
<pre><code>## MANIFEST.txt has been created at data/zip/MANIFEST.txt</code></pre>
</div>
<div id="merge-manual-datasets" class="section level2">
<h2>Merge Manual Datasets</h2>
<p>Datasets that have been manually annotated using the MATLAB code from
the <a href="https://github.com/hsosik/ifcb-analysis">ifcb-analysis</a>
repository (Sosik and Olson 2007) can be merged using the <a href="../reference/ifcb_merge_manual.html"><code>ifcb_merge_manual</code></a>
function. This is a wrapper function of the <a href="../reference/ifcb_create_class2use.html"><code>ifcb_create_class2use</code></a>,
<a href="../reference/ifcb_replace_mat_values.html"><code>ifcb_replace_mat_values</code></a>
and <a href="../reference/ifcb_adjust_classes.html"><code>ifcb_adjust_classes</code></a>
functions.</p>
<p>In this example, two datasets from the Swedish west coast are
downloaded from the <a href="https://doi.org/10.17044/scilifelab.25883455.v3">SMHI IFCB
Plankton Image Reference Library (version 3)</a> (Torstensson et
al. 2024) and combined into a single dataset. Please note that these
datasets are large, and the downloading and merging processes may take
considerable time.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="co"># Define data directories</span></span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a>skagerrak_kattegat_dir <span class="ot">&lt;-</span> <span class="st">&quot;data_skagerrak_kattegat&quot;</span></span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a>tangesund_dir <span class="ot">&lt;-</span> <span class="st">&quot;data_tangesund&quot;</span></span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a>merged_dir <span class="ot">&lt;-</span> <span class="st">&quot;data_skagerrak_kattegat_tangesund_merged&quot;</span></span>
<span id="cb11-5"><a href="#cb11-5" tabindex="-1"></a></span>
<span id="cb11-6"><a href="#cb11-6" tabindex="-1"></a><span class="co"># Download and extract Skagerrak-Kattegat data in the data folder</span></span>
<span id="cb11-7"><a href="#cb11-7" tabindex="-1"></a><span class="fu">ifcb_download_test_data</span>(<span class="at">dest_dir =</span> skagerrak_kattegat_dir,</span>
<span id="cb11-8"><a href="#cb11-8" tabindex="-1"></a>                        <span class="at">figshare_article =</span> <span class="st">&quot;48158725&quot;</span>)</span>
<span id="cb11-9"><a href="#cb11-9" tabindex="-1"></a></span>
<span id="cb11-10"><a href="#cb11-10" tabindex="-1"></a><span class="co"># Download and extract Tångesund data in the data folder</span></span>
<span id="cb11-11"><a href="#cb11-11" tabindex="-1"></a><span class="fu">ifcb_download_test_data</span>(<span class="at">dest_dir =</span> tangesund_dir,</span>
<span id="cb11-12"><a href="#cb11-12" tabindex="-1"></a>                        <span class="at">figshare_article =</span> <span class="st">&quot;48158731&quot;</span>)</span>
<span id="cb11-13"><a href="#cb11-13" tabindex="-1"></a></span>
<span id="cb11-14"><a href="#cb11-14" tabindex="-1"></a><span class="co"># Initialize the python session if not already set up</span></span>
<span id="cb11-15"><a href="#cb11-15" tabindex="-1"></a>env_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="fu">tempdir</span>(), <span class="st">&quot;iRfcb&quot;</span>) <span class="co"># Or your preferred venv path</span></span>
<span id="cb11-16"><a href="#cb11-16" tabindex="-1"></a><span class="fu">ifcb_py_install</span>(<span class="at">envname =</span> env_path)</span>
<span id="cb11-17"><a href="#cb11-17" tabindex="-1"></a></span>
<span id="cb11-18"><a href="#cb11-18" tabindex="-1"></a><span class="co"># Merge Skagerrak-Kattegat and Tångesund to a single dataset</span></span>
<span id="cb11-19"><a href="#cb11-19" tabindex="-1"></a><span class="fu">ifcb_merge_manual</span>(</span>
<span id="cb11-20"><a href="#cb11-20" tabindex="-1"></a>  <span class="at">class2use_file_base =</span> <span class="fu">file.path</span>(skagerrak_kattegat_dir, <span class="st">&quot;config/class2use.mat&quot;</span>),</span>
<span id="cb11-21"><a href="#cb11-21" tabindex="-1"></a>  <span class="at">class2use_file_additions =</span> <span class="fu">file.path</span>(tangesund_dir, <span class="st">&quot;config/class2use.mat&quot;</span>),</span>
<span id="cb11-22"><a href="#cb11-22" tabindex="-1"></a>  <span class="at">class2use_file_output =</span> <span class="fu">file.path</span>(merged_dir, <span class="st">&quot;config/class2use.mat&quot;</span>),</span>
<span id="cb11-23"><a href="#cb11-23" tabindex="-1"></a>  <span class="at">manual_folder_base =</span> <span class="fu">file.path</span>(skagerrak_kattegat_dir, <span class="st">&quot;manual&quot;</span>),</span>
<span id="cb11-24"><a href="#cb11-24" tabindex="-1"></a>  <span class="at">manual_folder_additions =</span> <span class="fu">file.path</span>(tangesund_dir, <span class="st">&quot;manual&quot;</span>),</span>
<span id="cb11-25"><a href="#cb11-25" tabindex="-1"></a>  <span class="at">manual_folder_output =</span> <span class="fu">file.path</span>(merged_dir, <span class="st">&quot;manual&quot;</span>))</span></code></pre></div>
<p>This concludes this tutorial for the <code>iRfcb</code> package. For
more detailed information, refer to the package documentation or the
other <a href="../articles/index.html">tutorials</a>. See how data
pipelines can be constructed using <code>iRfcb</code> in the following
<a href="https://github.com/nodc-sweden/ifcb-data-pipeline">Example
Project</a>. Happy analyzing!</p>
</div>
<div id="citation" class="section level2">
<h2>Citation</h2>
<pre><code>## To cite package &#39;iRfcb&#39; in publications use:
## 
##   Anders Torstensson (2025). iRfcb: Tools for Managing Imaging
##   FlowCytobot (IFCB) Data. R package version 0.4.3.
##   https://doi.org/10.5281/zenodo.12533225
## 
## A BibTeX entry for LaTeX users is
## 
##   @Manual{,
##     title = {iRfcb: Tools for Managing Imaging FlowCytobot (IFCB) Data},
##     author = {Anders Torstensson},
##     year = {2025},
##     note = {R package version 0.4.3},
##     url = {https://doi.org/10.5281/zenodo.12533225},
##   }</code></pre>
</div>
<div id="references" class="section level2">
<h2>References</h2>
<ul>
<li>Kraft, K., Velhonoja, O., Seppälä, J., Hällfors, H., Suikkanen, S.,
Ylöstalo, P., Anglès, S., Kielosto, S., Kuosa, H., Lehtinen, S., Oja,
J., Tamminen, T. (2022). SYKE-plankton_IFCB_2022 [Data set]. <a href="https://b2share.eudat.eu" class="uri">https://b2share.eudat.eu</a>. <a href="https://doi.org/10.23728/b2share.abf913e5a6ad47e6baa273ae0ed6617a" class="uri">https://doi.org/10.23728/b2share.abf913e5a6ad47e6baa273ae0ed6617a</a></li>
<li>Sosik, H. M. and Olson, R. J. (2007) Automated taxonomic
classification of phytoplankton sampled with imaging-in-flow cytometry.
Limnol. Oceanogr: Methods 5, 204–216.</li>
<li>Torstensson, A., Skjevik, A-T., Mohlin, M., Karlberg, M. and
Karlson, B. (2024). SMHI IFCB Plankton Image Reference Library.
SciLifeLab. Dataset. <a href="https://doi.org/10.17044/scilifelab.25883455.v3" class="uri">https://doi.org/10.17044/scilifelab.25883455.v3</a></li>
</ul>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
